src/utils/gpxUtils.js

```diff
@@ -41,7 +41,7 @@ export function parseGpx(gpxData, projection) {
         featureProjection: projection.epsg,
     })
     features.forEach((feature) => {
-        feature.setStyle(gpxStyle)
+        feature.setStyle(gpxStyles[feature.getGeometry().getType()] || gpxStyle)
```

Comment:
I would remove the fallback and make sure that the `gpxStyles` contain all geometry types. Then I would also remove the `gpxStyle` for better code clarity